Mathlib/AlgebraicGeometry/EllipticCurve/DivisionPolynomial/Basic.lean

```diff
@@ -15,75 +15,70 @@ These are defined in terms of the auxiliary sequences for normalised elliptic di
+
+Now, in the coordinate ring `R[W]`, note that `ψ₂²` is congruent to the polynomial
+`Ψ₂Sq := 4X³ + b₂X² + 2b₄X + b₆ ∈ R[X]`. As such, the recurrences of a normalised EDS show that
+`ψₙ / ψ₂` are congruent to certain polynomials in `R[W]`. In particular, define `preΨₙ ∈ R[X]` as
+the auxiliary sequence for a normalised EDS with extra parameter `Ψ₂Sq²` and initial values
+ * `preΨ₀ := 0`,
+ * `preΨ₁ := 1`,
+ * `preΨ₂ := 1`,
+ * `preΨ₃ := ψ₃`, and
+ * `preΨ₄ := ψ₄ / ψ₂`.
+
+The corresponding normalised EDS `Ψₙ ∈ R[X, Y]` is then given by
+ * `Ψₙ := preΨₙ ⬝ ψ₂` if `n` is even, and
+ * `Ψₙ := preΨₙ` if `n` is odd.
+
+Furthermore, define the associated sequences `ΨSqₙ, Φₙ ∈ R[X]` by
+ * `ΨSqₙ := preΨₙ² ⬝ Ψ₂Sq` if `n` is even,
+ * `ΨSqₙ := preΨₙ²` if `n` is odd,
+ * `Φₙ := XΨSqₙ - preΨₙ₊₁ ⬝ preΨₙ₋₁` if `n` is even, and
+ * `Φₙ := XΨSqₙ - preΨₙ₊₁ ⬝ preΨₙ₋₁ ⬝ Ψ₂Sq` if `n` is odd.
+
+With these definitions, `ψₙ ∈ R[X, Y]` and `φₙ ∈ R[X, Y]` are congruent in `R[W]` to `Ψₙ ∈ R[X, Y]`
+and `Φₙ ∈ R[X]` respectively, which are defined in terms of `Ψ₂Sq ∈ R[X]` and `preΨₙ ∈ R[X]`.
 
 ## Main definitions
 
- * `WeierstrassCurve.preΨ`: the univariate polynomials $\tilde{\Psi}_n$.
- * `WeierstrassCurve.ΨSq`: the univariate polynomials $\Psi_n^{[2]}$.
- * `WeierstrassCurve.Ψ`: the bivariate polynomials $\Psi_n$.
- * `WeierstrassCurve.Φ`: the univariate polynomials $\Phi_n$.
- * `WeierstrassCurve.ψ`: the bivariate $n$-division polynomials $\psi_n$.
- * `WeierstrassCurve.φ`: the bivariate polynomials $\phi_n$.
- * TODO: the bivariate polynomials $\omega_n$.
+ * `WeierstrassCurve.preΨ`: the univariate polynomials `preΨₙ`.
+ * `WeierstrassCurve.ΨSq`: the univariate polynomials `ΨSqₙ`.
+ * `WeierstrassCurve.Ψ`: the bivariate polynomials `Ψₙ`.
+ * `WeierstrassCurve.Φ`: the univariate polynomials `Φₙ`.
+ * `WeierstrassCurve.ψ`: the bivariate `n`-division polynomials `ψₙ`.
+ * `WeierstrassCurve.φ`: the bivariate polynomials `φₙ`.
+ * TODO: the bivariate polynomials `ωₙ`.
 
 ## Implementation notes
 
 Analogously to `Mathlib.NumberTheory.EllipticDivisibilitySequence`, the bivariate polynomials
-$\Psi_n$ are defined in terms of the univariate polynomials $\tilde{\Psi}_n$. This is done partially
-to avoid ring division, but more crucially to allow the definition of $\Psi_n^{[2]}$ and $\Phi_n$ as
-univariate polynomials without needing to work under the coordinate ring, and to allow the
-computation of their leading terms without ambiguity. Furthermore, evaluating these polynomials at a
-rational point on $W$ recovers their original definition up to linear combinations of the
-Weierstrass equation of $W$, hence also avoiding the need to work in the coordinate ring.
+`Ψₙ` are defined in terms of the univariate polynomials `preΨₙ`. This is done partially to avoid
+ring division, but more crucially to allow the definition of `ΨSqₙ` and `Φₙ` as univariate
+polynomials without needing to work under the coordinate ring, and to allow the computation of their
```

Comment:
What do you mean by "working under the coordinate ring"?

Reply:
I mean that they are actually bivariate polynomials, but if we quotient out by the equation of the curve (hence working in the coordinate ring rather than the full polynomial ring), there are univariate representatives that are equivalent to them. On the other hand, I can't actually define these univariate representatives in terms of `Ψₙ` because it's already bivariate, so I have to first define the univariate `preΨₙ` to define `ΨSqₙ` and `Φₙ`.

This is one of the things somewhat awkward to explain in docstrings (I wish we have the paper ready!).

Reply:
OK, thanks.